modules/user/user.service.js

```diff
@@ -13,15 +13,15 @@ class UserService {
     return User.find();
   }
 
-  async getUserById(id, auth) {
-    const user = await User.findById(id);
+  async getUserById(auth) {
+    const user = await User.findById(auth.userId);
```

Comment:
`checkUserExist` performs database request too. We should have only one db request to get user. You can refactor `checkUserExist` to `getUserOrThrow` and put it as a service method. In `utils` we should not have any functions that perform requests to business entities

Reply:
fixed